fhd_core/gridding/beam_per_baseline.pro

```diff
@@ -0,0 +1,62 @@
+function beam_per_baseline, psf, uu, vv, ww, l_mode, m_mode, n_tracked, frequency_array, x, y,$
+    xmin_use, ymin_use, freq_i, bt_index, polarization, fbin, image_bot, image_top, psf_dim3,$
+    box_matrix, vis_n, beam_int=beam_int, beam2_int=beam2_int, n_grp_use=n_grp_use,$
+    degrid_flag=degrid_flag
+
+;Make the beams on the fly with corrective phases given the baseline location. 
+;Will need to be rerun for every baseline, so speed is key.
+;For more information, see Jack Line's thesis
+
+;Loop over all visibilities that fall within the chosen visibility box
+FOR ii=0L,vis_n-1 DO begin
+    ;Pixel center offset phases
+    deltau_l = l_mode*(uu[bt_index[ii]]*frequency_array[freq_i[ii]]-x[xmin_use+psf.dim/2])
+    deltav_m = m_mode*(vv[bt_index[ii]]*frequency_array[freq_i[ii]]-y[ymin_use+psf.dim/2])
+    ;w term offset phase
+    w_n_tracked = n_tracked*ww[bt_index[ii]]*frequency_array[freq_i[ii]]
+
+    ;Generate a UV beam from the image space beam, offset by calculated phases
+    psf_base_superres=dirty_image_generate((*psf.image_power_beam_arr[polarization,fbin[ii]])*$
+      exp(2.*!pi*Complex(0,1)*(-w_n_tracked+deltau_l+deltav_m)),/no_real)
+ 
+    psf_base_superres=psf_base_superres[image_bot:image_top,image_bot:image_top]
+    d = size(psf_base_superres,/DIMENSIONS) & nx = d[0]/2 & ny = d[1]/2
+    ;A quick way to sum down the image by a factor of 2 in both dimensions.
+    ;  indices of all the 2x2 sub-arrays are next to each other in memory
+    ;  then, total collapses the 2x2 sub-arrays
+    psf_base_superres = transpose(total(reform(transpose(reform(psf_base_superres,2,nx,2*ny),$
```

Comment:
This would probably be faster as `psf_base_superres = Rebin(Temporary(psf_base_superres), nx, ny, /sample)`, but it would not be as accurate. You would also need to multiply by 4 if you wanted to preserve the sum.

Reply:
Could you add a comment explaining why you are resizing the image, and why you chose a factor of 2?